src/doc/rustc/src/codegen-options/index.md

```diff
@@ -71,9 +71,11 @@ If not specified, debug assertions are automatically enabled only if the
 This flag controls the generation of debug information. It takes one of the
 following values:
 
-* `0`: no debug info at all (the default).
-* `1`: line tables only.
-* `2`: full debug info.
+* `0` or `none`: no debug info at all (the default).
+* `line-directives-only`: line info directives only. For the nvptx* targets this enables [profiling](https://reviews.llvm.org/D46061). On other targets this behaves the same as `none`.
```

Comment:
@jdtatz do you know if this second sentence here is correct?

Reply:
This does not seem to be true. Looking at the output of `objdump --section-header` and `objdump -W`, there are a few differences between `none`, `line-directives-only`, and `line-tables-only`:
- `none` is the smallest and contains no debuginfo related sections
- `line-directives-only` contains the `.debug_line` section and (an empty) `.debug_aranges`.
- `line-tables-only` adds `.debug_abbrev`, `.debug_info`, `.debug_str`, and a proper `.debug_aranges` section.

So `line-directives-only` is a reduced (degraded?) version of `line-tables-only`. I could imagine that for most tools `line-directives-only` is effectively the same as `none`.

In terms of documentation, my suggestion would be 
```
* `line-directives-only`: line info directives only; a stripped down version of `line-tables-only`. For the nvptx* targets this enables [profiling](https://reviews.llvm.org/D46061). For other use cases, `line-tables-only` is the better, more compatible choice.
```